protos/flyteidl/core/types.proto

```diff
@@ -170,6 +170,26 @@ message OutputReference {
 
     // Variable name must refer to an output variable for the node.
     string var = 2;
+
+    repeated PromiseAtrribute attr_path = 3;
```

Comment:
would it be better to name it `attr`? btw, could you add a comment?

Reply:
should we just add an attribute and item in the `OutputReference`?
like
```proto
...
string attribute = 3;
string item = 4;
int index = 5;
```

Reply:
but attribute/item are list. Or you mean putting the above in `PromiseAtrribute`?
And in `OutputReference`, we still use 

```
repeated PromiseAtrribute
```

I think the idea to differentiate the attr to these 3 types is cleaner and more easier to understand

Reply:
After the offline discussion. We will keep PromiseAttribute fields as is